src/abstractions/MultipartBody.cs

```diff
@@ -47,10 +47,11 @@ public void AddOrReplacePart<T>(string partName, string contentType, T partValue
         {
             throw new ArgumentNullException(nameof(partValue));
         }
+        var key = Tuple.Create(partName, fileName);
```

Comment:
do you think we could use value tuples instead?
It's a small change but should reduce allocations.
https://learn.microsoft.com/en-us/dotnet/api/system.valuetuple?view=net-9.0